diff --git a/dist/cjs/shared/transport.d.ts b/dist/cjs/shared/transport.d.ts
index 1bd95efb41fcfd1ae09712ca968c57d828e0acaf..6b37b96de70ea974e8734878787d005a51761937 100644
--- a/dist/cjs/shared/transport.d.ts
+++ b/dist/cjs/shared/transport.d.ts
@@ -62,13 +62,13 @@ export interface Transport {
      *
      * This should be invoked when close() is called as well.
      */
-    onclose?: () => void;
+    onclose?: (() => void) | undefined;
     /**
      * Callback for when an error occurs.
      *
      * Note that errors are not necessarily fatal; they are used for reporting any kind of exceptional condition out of band.
      */
-    onerror?: (error: Error) => void;
+    onerror?: ((error: Error) => void) | undefined;
     /**
      * Callback for when a message (request or response) is received over the connection.
      *
@@ -76,14 +76,14 @@ export interface Transport {
      *
      * The requestInfo can be used to get the original request information (headers, etc.)
      */
-    onmessage?: <T extends JSONRPCMessage>(message: T, extra?: MessageExtraInfo) => void;
+    onmessage?: (<T extends JSONRPCMessage>(message: T, extra?: MessageExtraInfo) => void) | undefined;
     /**
      * The session ID generated for this connection.
      */
-    sessionId?: string;
+    sessionId?: string | undefined;
     /**
      * Sets the protocol version used for the connection (called when the initialize response is received).
      */
-    setProtocolVersion?: (version: string) => void;
+    setProtocolVersion?: ((version: string) => void) | undefined;
 }
 //# sourceMappingURL=transport.d.ts.map
\ No newline at end of file
diff --git a/dist/esm/shared/transport.d.ts b/dist/esm/shared/transport.d.ts
index 1bd95efb41fcfd1ae09712ca968c57d828e0acaf..6b37b96de70ea974e8734878787d005a51761937 100644
--- a/dist/esm/shared/transport.d.ts
+++ b/dist/esm/shared/transport.d.ts
@@ -62,13 +62,13 @@ export interface Transport {
      *
      * This should be invoked when close() is called as well.
      */
-    onclose?: () => void;
+    onclose?: (() => void) | undefined;
     /**
      * Callback for when an error occurs.
      *
      * Note that errors are not necessarily fatal; they are used for reporting any kind of exceptional condition out of band.
      */
-    onerror?: (error: Error) => void;
+    onerror?: ((error: Error) => void) | undefined;
     /**
      * Callback for when a message (request or response) is received over the connection.
      *
@@ -76,14 +76,14 @@ export interface Transport {
      *
      * The requestInfo can be used to get the original request information (headers, etc.)
      */
-    onmessage?: <T extends JSONRPCMessage>(message: T, extra?: MessageExtraInfo) => void;
+    onmessage?: (<T extends JSONRPCMessage>(message: T, extra?: MessageExtraInfo) => void) | undefined;
     /**
      * The session ID generated for this connection.
      */
-    sessionId?: string;
+    sessionId?: string | undefined;
     /**
      * Sets the protocol version used for the connection (called when the initialize response is received).
      */
-    setProtocolVersion?: (version: string) => void;
+    setProtocolVersion?: ((version: string) => void) | undefined;
 }
 //# sourceMappingURL=transport.d.ts.map
